httphandlers/http_handlers.go

```diff
@@ -29,26 +35,42 @@ func (h httpMetricsHandler) ServeHTTP(w http.ResponseWriter, req *http.Request)
 	t.Time(func() { h.handler.ServeHTTP(w, req) })
 }
 
-// TransactionAwareRequestLoggingHandler finds a transactionID on a request header or generates one, and makes sure it gets logged
-// using the supplied UPP logger
-func TransactionAwareRequestLoggingHandler(logger *logger.UPPLogger, h http.Handler) http.Handler {
-	return transactionAwareRequestLoggingHandler{logger, h}
+type handlerOpt func(h *transactionAwareRequestLoggingHandler)
+
+// FilterHeaders creates a handler option that extends denied header list.
+// When creating the log entry the handler log the request headers. For clarity and security some of the headers are filtered out by default.
+// The default filter list could be extended by providing list of denied header keys.
+func FilterHeaders(headers []string) handlerOpt { // nolint:golint // we don't want handlerOpt exported
+	return func(h *transactionAwareRequestLoggingHandler) {
+		h.deniedHeaders = headers
+	}
+}
+
+// TransactionAwareRequestLoggingHandler creates new http.Handler that would add log entries to the provided logger in structured format.
+// The handler would search for transactionID in the request headers and will generate one if it doesn't find any.
+func TransactionAwareRequestLoggingHandler(log *logger.UPPLogger, handler http.Handler, options ...handlerOpt) http.Handler {
+	h := transactionAwareRequestLoggingHandler{logger: log, handler: handler, deniedHeaders: nil}
+	for _, opt := range options {
+		opt(&h)
+	}
+	return h
 }
 
 type transactionAwareRequestLoggingHandler struct {
-	logger  *logger.UPPLogger
-	handler http.Handler
+	logger        *logger.UPPLogger
+	handler       http.Handler
+	deniedHeaders []string
 }
 
 func (h transactionAwareRequestLoggingHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	transactionID := transactionidutils.GetTransactionIDFromRequest(req)
 	w.Header().Set(transactionidutils.TransactionIDHeader, transactionID)
+	req.Header.Set(transactionidutils.TransactionIDHeader, transactionID)
```

Comment:
Looking at the implementation of `transactionidutils.GetTransactionIDFromRequest` I think we could skip setting the header here.

Reply:
Removed it